File: api/src/interfaces/paging/pagination.ts
export const DEFAULT_PAGINATION_REQUEST: IPaginationRequest = {
  pageIndex: 1,
  pageSize: 10,
};

export interface IPaginationRequest {
  pageIndex: number;
  pageSize: number;
  offset?: number;
}

export const DEFAULT_PAGINATION_RESPONSE: IPaginationResponse = {
  pageCount: 1,
};

export interface IPaginationResponse {
  pageCount: number;
}

File: back/product-service/pkg/models/pagination.go
package models

const DEFAULT_PAGE_SIZE = 10
const DEFAULT_PAGE_INDEX = 1

type PaginationRequest struct {
	PageIndex uint	`json:"page_index" validate:"required"`
	PageSize 	uint	`json:"page_size" validate:"required"`
	Offset		uint	`json:"offset"`
}

type PaginationResponse struct {
	PageCount	uint	`json:"page_count"`
}
